src/osd/PG.cc

```diff
@@ -5817,6 +5817,7 @@ void PG::handle_loaded(RecoveryCtx *rctx)
   dout(10) << "handle_loaded" << dendl;
   Load evt;
   recovery_state.handle_event(evt, rctx);
+  write_if_dirty(*rctx->transaction);
```

Comment:
i am not sure if this works. in `OSD::load_pgs()`, a nullptr is passed in as the transaction of rctx. like
```c++
    PG::RecoveryCtx rctx(0, 0, 0, 0, 0, 0);
    pg->handle_loaded(&rctx);
```
can we dereference it here w/o checking it beforehand?

Reply:
fixed this one already